extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -360,10 +360,9 @@ export class AmpLightboxGallery extends AMP.BaseElement {
     return this.mutateElement(() => {
       const {length} = this.elementsMetadata_[lightboxGroupId];
       this.maybeEnableMultipleItemControls_(length);
-      this.carousel_.getImpl().then((implementation) => {
-        implementation.unlayoutCallback();
```

Comment:
This change hasn't made it to production yet I believe, are you sure this is the root cause for the production errors?

Reply:
Interesting. This patch fixes the issue, although the issue predates this addition.

Reply:
I did a bisect to find the original problem commit. It is fun 

- The issue was introduced via https://github.com/ampproject/amphtml/pull/32471. Switching to ResizeObserver created a race condition where the `slideWidth_` wasn't updated to a nonzero value in time to change slides.
- The addition of `unlayoutCallback` (https://github.com/ampproject/amphtml/pull/34539) exacerbates the issue. Even if the earlier change is reverted, the issue remains.

Luckily, just addressing the second point seems to work.